### ISSUE TITLE
BatchPacket is already encoded

### ISSUE THREAD

**Issue by SOF3** [Falkirks/Specter#47]
ChunkRequestTask creates the BatchPacket without filling the payload field, causing a crash.

@dktapps This might be an API problem, although I understand that it improves performance.

```
s s alice
[21:04:35] [Server thread/INFO]: alice[/SPECTER:19133] logged in with entity id 1 at (world, 256, 72, 256)
[21:04:35] [Server thread/INFO]: Session started.
[21:04:35] [Server thread/CRITICAL]: ErrorException: "zlib_decode(): data error" (EXCEPTION) in "/plugins/Specter.phar/src/specter/network/SpecterInterface" at line 117
[21:04:35] [Server thread/DEBUG]: #0 (): pocketmine\{closure}(integer 2, string zlib_decode(): data error, string phar://C:/sofe/pmrun/plugins/Specter.phar/src/specter/network/SpecterInterface.php, integer 117, array Array())
[21:04:35] [Server thread/DEBUG]: #1 /plugins/Specter.phar/src/specter/network/SpecterInterface(117): zlib_decode(string , integer 67108864)
[21:04:35] [Server thread/DEBUG]: #2 /src/pocketmine/Player(1084): specter\network\SpecterInterface->putPacket(specter\network\SpecterPlayer object, pocketmine\network\mcpe\protocol\BatchPacket object, boolean , boolean )
[21:04:35] [Server thread/DEBUG]: #3 /src/pocketmine/Player(817): pocketmine\Player->dataPacket(pocketmine\network\mcpe\protocol\BatchPacket object)
[21:04:35] [Server thread/DEBUG]: #4 /src/pocketmine/level/Level(2415): pocketmine\Player->sendChunk(integer 16, integer 16, pocketmine\network\mcpe\protocol\BatchPacket object)
[21:04:35] [Server thread/DEBUG]: #5 /src/pocketmine/level/format/io/ChunkRequestTask(90): pocketmine\level\Level->chunkRequestCallback(integer 16, integer 16, pocketmine\network\mcpe\protocol\BatchPacket object)
[21:04:35] [Server thread/DEBUG]: #6 /src/pocketmine/scheduler/AsyncPool(160): pocketmine\level\format\io\ChunkRequestTask->onCompletion(pocketmine\Server object)
[21:04:35] [Server thread/DEBUG]: #7 /src/pocketmine/scheduler/ServerScheduler(341): pocketmine\scheduler\AsyncPool->collectTasks(boolean)
[21:04:35] [Server thread/DEBUG]: #8 /src/pocketmine/Server(2422): pocketmine\scheduler\ServerScheduler->mainThreadHeartbeat(integer 167)
[21:04:35] [Server thread/DEBUG]: #9 /src/pocketmine/Server(2190): pocketmine\Server->tick(boolean)
[21:04:35] [Server thread/DEBUG]: #10 /src/pocketmine/Server(2072): pocketmine\Server->tickProcessor(boolean)
[21:04:35] [Server thread/DEBUG]: #11 /src/pocketmine/Server(1654): pocketmine\Server->start(boolean)
[21:04:35] [Server thread/DEBUG]: #12 /src/pocketmine/PocketMine(501): pocketmine\Server->__construct(BaseClassLoader object, pocketmine\utils\MainLogger object, string C:\sofe\pmrun\, string C:\sofe\pmrun\, string C:\sofe\pmrun\plugins\)
```

using Falkirks/Specter@9cc3ab1beca6ffe21e6f7ab31a7d34ed89d204de and pmmp/PocketMine-MP@bf3868c078444a37629bfab231b5f4bed358a13c

**Comment by dktapps** [comment]
`$packet->decode()` before trying to read the payload should solve the problem... you should decode anyway as if this was a normal client.

**Comment by SOF3** [comment]
Fixed in #48
